Simplenote/NoteEditorViewController.m

```diff
@@ -1160,4 +1147,25 @@ - (NSPopover *)newPopoverWithContentViewController:(NSViewController *)viewContr
     return popover;
 }
 
+- (BOOL)urlSchemeIsAllowed: (NSString *) scheme {
+    return [scheme isEqualToString:@"http"] ||
+        [scheme isEqualToString:@"https"] ||
+        [scheme isEqualToString:@"mailto"];
+}
+
+#pragma mark - WKNavigationDelegate
+
+- (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
+    if (navigationAction.navigationType == WKNavigationTypeLinkActivated) {
+        NSURL *linkUrl = navigationAction.request.URL;
+        if ([self urlSchemeIsAllowed:linkUrl.scheme]) {
+            [[NSWorkspace sharedWorkspace] openURL:linkUrl];
+        }
+        
+        decisionHandler(WKNavigationActionPolicyCancel);
```

Comment:
We should probably bail out once the handler is called. Can we add a `return` here?